src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -390,5 +401,17 @@ public virtual async Task<VersionStamp> GetTopLevelChangeTextVersionAsync(Cancel
             TextAndVersion textAndVersion = await this.textAndVersionSource.GetValueAsync(cancellationToken).ConfigureAwait(false);
             return textAndVersion.Version;
         }
+
+        internal sealed class DefaultDocumentServiceProvider : IDocumentServiceProvider
+        {
+            public static readonly DefaultDocumentServiceProvider Instance = new DefaultDocumentServiceProvider();
+
+            private DefaultDocumentServiceProvider() { }
+
+            public TService GetService<TService>() where TService : class, IDocumentService
+            {
+                return default;
```

Comment:
this seems unfortunate.  Does that mean clients need to be aware that this may just fail in the *common* case?  Would it not be better to emit no-op services here so that clients can *always* defer work to an IDocumentServiceProvider?

In other words, tying this into your PR for FindRefs, wouldn't this mean FindsRefs would need a fallback path to handle this not being here?  Wouldn't it be better for features if htey didn't need the fallback path and could just trust that they could get correct behavior for their documents whenever they called into the document service provider?